### PR TITLE
[FEATURE] Use RTE in replacement field

### PR DESCRIPTION
**Pull request by jrenggli**

### CODE REVIEW

File: Configuration/TCA/tx_variables_marker.php [hunk 1]
```diff
@@ -136,6 +136,7 @@
                 'rows' => 4,
                 'eval' => 'required,trim'
             ],
+            'defaultExtras' => 'richtext:rte_transform[mode=css]',
```
Comment (xperseguers) [inline Configuration/TCA/tx_variables_marker.php]: official code (in TYPO3 v7) would be

```
'defaultExtras' => 'richtext:rte_transform[mode=ts_css]',
```

Comment (jrenggli) [reply Configuration/TCA/tx_variables_marker.php]: @xperseguers: Thanks for your response. I just adjusted that... and learnt something new

**Comment by xperseguers** [comment]
Just for the memo, this effectively allow "some" formatting to be applied to the replacement part, but it does not properly handle links, images, ... since those formatting would require frontend/TS processing on the content coming from the DB.

**Review by fanat98 — APPROVED** [review]

File: Configuration/TCA/tx_variables_marker.php [hunk 2]
```diff
@@ -136,6 +143,7 @@
                 'rows' => 4,
                 'eval' => 'required,trim'
             ],
+            'defaultExtras' => $settings['enableRte']?'richtext:rte_transform[mode=ts_css]':'',
```
Comment (fabarea) [inline Configuration/TCA/tx_variables_marker.php]:
```
$settings['enableRte']
```
May not be an existing index. I know TYPO3 will not complain but nevertheless we should avoid that IMO:

+ PSR-2 formatting needed:

```
$enableRte ? 'richtext:rte_transform[mode=ts_css]' : '',
```

File: Configuration/TCA/tx_variables_marker.php [hunk 3]
```diff
@@ -1,4 +1,11 @@
 <?php
+
+$settings = unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['variables']);
```
Comment (fabarea) [inline Configuration/TCA/tx_variables_marker.php]:
```
$settings = isset($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['variables']) 
    ? unserialize($GLOBALS['TYPO3_CONF_VARS']['EXT']['extConf']['variables'])
    : []
```

File: Configuration/TCA/tx_variables_marker.php [hunk 4]
```diff
@@ -1,4 +1,11 @@
...
+
+if (!is_array($settings)) {
+    $settings = [];
+}
+
```
Comment (fabarea) [inline Configuration/TCA/tx_variables_marker.php]:
```
$enableRte = is_array($settings) && isset($settings['enableRte']) 
   ? (bool)$settings['enableRte']
   : false
```

**Review by fabarea — CHANGES_REQUESTED** [review]

**Review by fabarea — APPROVED** [review]

**Comment by jrenggli** [comment]
Thanks for all your inputs...

**Pull request merged**